packages/turbo-telemetry/src/client.ts

```diff
@@ -147,17 +147,135 @@ export class TelemetryClient {
     }
   }
 
+  private trackCliOption({
+    option,
+    value,
+  }: {
+    option: string;
+    value: string;
+  }): Event {
+    return this.track({
+      key: `option:${option}`,
+      value,
+    });
+  }
+
+  private trackCliArgument({
+    argument,
+    value,
+  }: {
+    argument: string;
+    value: string;
+  }): Event {
+    return this.track({
+      key: `argument:${argument}`,
+      value,
+    });
+  }
+
+  private trackCliCommand({
+    command,
+    value,
+  }: {
+    command: string;
+    value: string;
+  }): Event {
+    return this.track({
+      key: `command:${command}`,
+      value,
+    });
+  }
+
   ////////////
   // EVENTS //
   ////////////
 
-  /**
-   * Track selected package manager
-   */
-  trackPackageManager(packageManager: string): Event {
-    return this.track({
-      key: "package_manager",
-      value: packageManager,
+  trackCommandStatus({
+    command,
+    status,
+  }: {
+    command: string;
+    status: string;
+  }): Event {
+    return this.trackCliCommand({
+      command,
+      value: status,
     });
   }
+
+  // CLI Options
+
+  trackOptionExample(value: string | undefined): Event | undefined {
+    if (value) {
+      return this.trackCliOption({
+        option: "example",
+        value,
+      });
+    }
+  }
+
+  trackOptionPackageManager(value: string | undefined): Event | undefined {
+    if (value) {
+      return this.trackCliOption({
+        option: "package_manager",
+        value,
+      });
+    }
+  }
+
+  trackOptionSkipInstall(value: boolean | undefined): Event | undefined {
```

Comment:
all these new methods seem really specific to create-turbo, not sure they should be in the telemtry package. can it be more generic?

Reply:
They're intentionally specific to allow auditing of events. It's the same approach we took for the core telemetry, but yea could subclass this per package to organize them a bit more.

Reply:
@mehulkar updated to have package specific event classes, this keeps things a bit more organized.

 Ideally we could just have super generic events that a re-used everywhere. But we need to have the events defined:

1. In a single location (not spread out across the code base)
2. As static as possible (no generic key=value)

This ensures that they are easily publicly auditable. 

Let me know what you think!

Reply:
Relevant commit: https://github.com/vercel/turbo/pull/7852/commits/a4951541922e74cad09d2c45fc708e445a829690